docs/high-availability.md

```diff
@@ -0,0 +1,199 @@
+the simple [Docker based single node cluster instructions](https://github.com/GoogleCloudPlatform/kubernetes/blob/master/docs/getting-started-guides/docker.md),
+or try [Google Container Engine](https://cloud.google.com/container-engine/) for hosted Kubernetes.
+
+Also, at this time high availability support for Kubernetes is not continuously tested in our end-to-end (e2e) testing.  We will
+be working to add this continuous testing, but for now the single-node master installations are more heavily tested.
+
+## Overview
+Setting up a truly reliable, highly available distributed system requires a number of steps, it is akin to
+wearing underwear, pants, a belt, suspenders, another pair of underwear, and another pair of pants.  We go into each
+of these steps in detail, but a summary is given here to help guide and orient the user.
+
+The steps involved are as follows:
+   * Installing a health-checking daemon runner
+   * Setting up a redundant, reliable storage layer with clustered etcd.
+   * Starting replicated, load balanced Kubernetes API servers
+   * Setting up master-elected Kubernetes scheduler and controller-manager daemons
+
+Here's what the system should look like when it's finished:
+![High availability Kubernetes diagram](high-availability/ha.png)
+
+Ready? Let's get started.
+
+## Initial set-up.
+The remainder of this guide assumes that you are setting up a 3-node clustered master, where each machine is running some flavor of Linux.
+Examples in the guide are given for Debian distributions, but they should be easily adaptable to other distributions.
+Likewise, this set up should work whether you are running in a public or private cloud provider, or if you are running
+on bare metal.
+
+The easiest way to implement an HA Kubernetes cluster is to start with an existing single-master cluster.  The
+instructions at [https://get.k8s.io](https://get.k8s.io)
+describe easy installation for single-master clusters on a variety of platforms.
+
+## Reliable nodes
+On each master node, we are going to run a number of processes that implement the Kubernetes API.  The first step in making these reliable is
+to make sure that each automatically restarts when it fails.  To achieve this, we need to install a process watcher.  We choose to use
+the ```kubelet``` that we use on each of the nodes.  This is convenient, since we can use containers to distribute our binaries, we can
+establish resource limits, and introspect the resource usage of each daemon.  Of course, we also need something to monitor the kubelet
+itself (insert who watches the watcher jokes here).  For Debian systems, we choose monit, but there are a number of alternate
+choices. For example, on systemd-based systems (e.g. RHEL, CentOS), you can run 'systemctl enable kubelet'. 
+
+You should install the [kubelet binary](https://storage.googleapis.com/kubernetes-release/release/v0.18.0/bin/linux/amd64/kubelet), the
```

Comment:
Mention that this should already be running on the master node if they did a regular single-node setup?